cozmo_soar.py

```diff
@@ -648,6 +654,153 @@ def on_input_phase(self, input_link: sml.Identifier):
                 status_wme.update_wm()
                 self.actions.remove((action, status_wme, root_id))
 
+        #######
+        # SVS #
+        #######
+        # obj1 comesfrom navmap, obj2 comes from buffer.  test if they are equal (disregarding small changes in vision)
+        def equal(obj1, obj2):
```

Comment:
This function should be defined somewhere else (perhaps a new navmap utility file). It doesn't really belong to the `CozmoSoar` object. It also shouldn't be defined in the `__init__` method, or in any other method, it should be a free-standing method of its own.

Name needs to be less generic, perhaps `blocks_equal`. `equal` functions generally provide equality checks for the object itself (in this case the `CozmoSoar` object).

```suggestion
        def blocks_equal(obj1, obj2):
```